agent/whatweb_agent.py

```diff
@@ -58,8 +69,32 @@ def process(self, message: msg.Message) -> None:
         """
         logger.info('processing message of selector : %s', message.selector)
         with tempfile.NamedTemporaryFile() as fp:
-            self._start_scan(message.data['name'], fp.name)
-            self._parse_emit_result(message.data['name'], fp)
+            target = self._prepare_target(message)
+            self._start_scan(target.domain, fp.name)
+            self._parse_emit_result(target.domain, fp, int(target.port), target.schema)
+
+    def _prepare_target(self, message: msg.Message) -> Target:
+        """Returns a target object to be scanned."""
+        if message.selector.endswith('.link'):
```

Comment:
As discussed, please only check for the fields present to make it easier to extend it in the future.